02.calendar/cal.rb

```diff
@@ -0,0 +1,145 @@
+require "date"
+require "optparse"
+
+# 数字表記の月：英語表記の月
+Month_name = {
+  "1" => "Jan", 
+  "2" => "Feb",
+  "3" => "Mar", 
+  "4" => "Apr", 
+  "5" => "May", 
+  "6" => "Jun", 
+  "7" => "Jul",
+  "8" => "Aug",
+  "9" => "Sep",
+  "10" => "Oct",
+  "11" => "Nov",
+  "12" => "Dec"
+}
+
+Week_date = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"]
+
+# ANSIエスケープシーケンス（背景：白、文字色：黒）
+ANSI_ESC = "\e[47m\e[30m"
+
+# ANSIエスケープシーケンスのクローズ
+ANSI_ESC_END = "\e[m"
+
+
+class Calendar
+
+  # 初期化
+  def initialize(year, month, day)
+    @year = year
+    @month = month
+    @day = day
+  end
+
+  # 月の最初の曜日を返す
+  def get_first_wday
+    return Date.new(@year, @month, 1).wday
+  end
+
+  # 月を数字表記から英語表記に直す
+  def convert_month_to_english(month)
+    # month_nameのハッシュデータのkeyを探索する
+    Month_name[month.to_s]
+  end
+
+  # カレンダーの曜日をコンソールに表示する
+  def print_all_weekdate
+    puts " " + Week_date.join(' ')
+  end
+
+  # カレンダーの日付を整形して表示する
+  #
+  # 引数
+  # wday: 指定した月の曜日の整数表記
+  # last_day: 表示する月の最終日(28 or 29 or 30 or 31)
+  def print_date(wday, last_day)
+
+    # 開始位置の設定
+    set_first_position(wday)
+  
+    (1..last_day).each do |i|
+   
+      # デフォルト出力フォーマット
+      print_date = i.to_s
+      
+      # 今日日付の場合
+      if today?(i)
+        print_date = "  " + ANSI_ESC + print_date + ANSI_ESC_END
+      end
```

Comment:
このままでもいいんですが、Rubyでは処理が一行で済む場合は後置ifが好まれますね。

```suggestion
      print_date = "  " + ANSI_ESC + print_date + ANSI_ESC_END if today?(i)
```